src/biokbase/narrative/jobs/job.py

```diff
@@ -187,32 +266,21 @@ def update_state(self, state=None):
                 f"Job ID mismatch in update_state: job ID: {self.job_id}; state ID: {state['job_id']}"
             )
 
-        # TODO: add in a check for the case where this is a batch parent job
-        # batch parent jobs with where all children have status "completed" are in a terminal state
-        # otherwise, child jobs may be retried
-        self.terminal_state = (
-            True if state.get("status", "unknown") in TERMINAL_STATUSES else False
-        )
-
-        # delete fields that we are not interested in
-        for field in EXCLUDED_JOB_STATE_FIELDS:
-            if field in state and field != "job_input":
-                del state[field]
-
+        state = copy.deepcopy(state)
         if self._last_state is None:
             self._last_state = state
         else:
             # TODO: implement batch job code updates here
+            # TODO: update state structure-wise
             self._last_state.update(state)
 
-        return self._last_state
+        return copy.deepcopy(self._last_state)
 
-    def reset_state(self):
+    def clear_state(self):
```

Comment:
Is this only used for testing? If there are any other methods/attributes like this, it's probably worth putting a comment in the method doc to note that.

Reply:
I kind of just wanted to define them in the test file instead

Reply:
I guess that could make it less widely accessible